opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -575,15 +586,8 @@ def set_attribute(self, key: str, value: types.AttributeValue) -> None:
             with self._lock:
                 self.attributes[key] = value
```

Comment:
What about `set_attribute`?

Reply:
Ah yeah good point, I saw the lock here and thought it was safe from this condition, but it is not. will fix

Reply:
Updated